configure.ac

```diff
@@ -995,6 +999,52 @@ AC_SUBST(incflags, "$INCFLAGS")
 test -z "${ac_env_CFLAGS_set}" -a -n "${cflags+set}" && eval CFLAGS="\"$cflags $ARCH_FLAG\""
 test -z "${ac_env_CXXFLAGS_set}" -a -n "${cxxflags+set}" && eval CXXFLAGS="\"$cxxflags $ARCH_FLAG\""
 
+# The lines above expand out the $cflags/$optflags/$debugflags/$hardenflags variables into the
+# CFLAGS variable. So, at this point, we have a $CFLAGS var with the actual compiler flags we're
+# going to use.
+# That means this is the right time to check what branch protection flags are going to be in use
+# and define appropriate macros for use in Context.S based on this
+AS_CASE(["$target_cpu"], [aarch64], [
+    AC_MSG_CHECKING(whether __ARM_FEATURE_BTI_DEFAULT is defined)
```

Comment:
Check every time, not using `AC_CACHE_CHECK`?

Reply:
I actually didn't think about caching. but, thinking about it now, it should be uncached - these defines will change depending on the cflags used. (Or does the autoconf cache invalidate if cflags are changed? I’ll try it tomorrow)

Reply:
Good news, looks like autoconf is smart enough to bust the cache here.

```
ktsanaktsidis@lima-fedora build % ../configure  --prefix=$(realpath ~/rubies/3.4.0-master) --disable-install-doc --disable-yjit cflags="-mbranch-protection=standard" --config-cache
configure: loading cache config.cache
configure: error: `cflags' has changed since the previous run:
configure:   former value:  `-mbranch-protection=none'
configure:   current value: `-mbranch-protection=standard'
configure: error: in `/home/ktsanaktsidis.linux/ruby/build':
configure: error: changes in the environment can compromise the build
configure: error: run `make distclean' and/or `rm config.cache'
	    and start over
```

So i'll wrap it in `AC_CACHE_CHECK`.